### PR TITLE
Feat: Huffmate V1

### PR DESCRIPTION
**Pull request by abigger87**
## Overview

This PR will release the first stable version of huffmate!

### Checklist

`tokens`
- [x] ERC1155
- [x] ERC721
- [x] ERC20
- [x] ERC4626

`mechanisms`
- [x] vrgdas
- [x] clones

`proxies`
- [x] Clones
- [x] Proxy

`utils`
- [x] Calls
- [x] Shuffling
- [x] Create3
- [x] SSTORE2
- [x] SafeTransferLib
- [x] Refunded

### Changelog

- Initial ERC1155 Implementation in #17 by @cheethas and @abigger87 
- ERC20 Implementation in #7 by @devtooligan and @abigger87 
- ERC721 Implementation in #20 by @kadenzipfel 
- Merkle Distributor in #47 by @benleim 
- Huff Clones in #49 by @clabby 
- ERC4626 in #48 by @abigger87 
- SafeTransferLib in #52 by @clabby 
- huff-vrgdas in #55 by @cheethas ported by @abigger87 
- SSTORE2 in #54 by @abigger87 
- CREATE3 in #59 by @cheethas and @abigger87 
- Shuffling in #56 by @abigger87 
- Refunded in #65 by @abigger87 
- Proxies and Minimal Clones in #61 by @cheethas and @abigger87 

### Room for Improvement

- Documenting
- Gas Golfing

### CODE REVIEW

**Comment by kadenzipfel** [comment]
started working on erc721 again! will be done asap

**Review by kadenzipfel — CHANGES_REQUESTED** [review]

File: src/tokens/ERC20.huff [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+    0x00 mstore                     // [to, to]
+    [ZERO_ADDRESS]                  // [address(0), to, to]
+    __EVENT_HASH(Transfer)          // [sig, from, to]
+    0x20 0x00                       // [0, 32, sig, from, to]
+    log3                            // []
+}
+
+
+// Function Dispatching
+#define macro MAIN_ERC20() = takes (1) returns (1) {
```
[the remark this answers is not in the document]
Comment (kadenzipfel) [reply src/tokens/ERC20.huff:538]: noticed this in a few other places as well. tbh both strategies are good but i think important to maintain consistency throughout the repo by defining which style we prefer

**Comment by Philogy** [comment]
I propose adding the shuffle macro I've written to mechanisms for V1: https://github.com/Philogy/erc721h/blob/main/src/utils/Mechs.huff

**Comment by abigger87** [comment]
> I propose adding the shuffle macro I've written to mechanisms for V1: https://github.com/Philogy/erc721h/blob/main/src/utils/Mechs.huff

Very cool, will add. Thanks!

**Pull request merged**